libsolidity/interface/FileReader.h

```diff
@@ -61,18 +61,17 @@ class FileReader
 	void allowDirectory(boost::filesystem::path _path);
 	FileSystemPathSet const& allowedDirectories() const noexcept { return m_allowedDirectories; }
 
-	StringMap const& sourceCodes() const noexcept { return m_sourceCodes; }
-
-	/// Retrieves the source code for a given source unit name.
-	SourceCode const& sourceCode(SourceUnitName const& _sourceUnitName) const { return m_sourceCodes.at(_sourceUnitName); }
+	/// @returns all sources by their internal source unit names.
+	StringMap const& sourceUnits() const noexcept { return m_sourceCodes; }
 
 	/// Resets all sources to the given map of source unit name to source codes.
 	/// Does not enforce @a allowedDirectories().
-	void setSources(StringMap _sources);
+	void setSourceUnits(StringMap _sources);
 
-	/// Adds the source code under a source unit name created by normalizing the file path.
+	/// Adds the source code under a source unit name created by normalizing the file path
+	/// or changes an existing source.
 	/// Does not enforce @a allowedDirectories().
-	void setSource(boost::filesystem::path const& _path, SourceCode _source);
+	void addOrUpdateFile(boost::filesystem::path const& _path, SourceCode _source);
```

Comment:
How about this?
```suggestion
	void loadSourceFromPath(boost::filesystem::path const& _path, SourceCode _source);
```

Reply:
This sounds like it would read from the filesystem.

Reply:
So `addSourceFromPath()`. I mean, I'd emphasize that the path is being translated here into a source unit name. We could also go for `addSourceFromCLI()`

Reply:
I will add it to the docs. I'm not sure we should emphasize the "CLI" aspect here. The only thing to consider here is that it is a path that is assumed to exist in the host filesystem.

Reply:
Ah, it already is part of the docstring.